app/src/main/java/com/kelvin/bdaypost/notifications/NotificationMessagingService.kt

```diff
@@ -0,0 +1,45 @@
+package com.kelvin.bdaypost.notifications
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.kelvin.bdaypost.util.SharedPrefsUtil
+import dagger.hilt.android.AndroidEntryPoint
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+@AndroidEntryPoint
+class NotificationMessagingService : FirebaseMessagingService() {
+
+    @Inject
+    lateinit var sharedPrefsUtil: SharedPrefsUtil
+
+    @Inject
+    lateinit var notificationSender: NotificationSender
+
+    /**
+     * Called if the FCM registration token is updated. This may occur if the security of
+     * the previous token had been compromised. Note that this is called when the
+     * FCM registration token is initially generated so this is where you would retrieve the token.
+     */
+    override fun onNewToken(token: String) {
+        Timber.d("Refreshed token: $token")
+        sharedPrefsUtil.putString(NOTIFICATION_KEY, token)
+    }
+
+    /**
+     * Called when app is in foreground. If app is in background, notification goes directly
```

Comment:
and what happens when they are in the foreground? lol

Reply:
It's whatever we put inside `onMessageReceived()`. So the only time we get to programatically configure the notification via `NotificationSender` is when the app is foregrounded. When it's in background, the notification is completely configured by Firebase console.